### PR TITLE
Refactor NullNode and remove NullLeafNode class

### PR DESCRIPTION
**Pull request by lu-pinto**
Remove NullLeafNode and use NullNode instead

### CODE REVIEW

File: src/main/java/org/hyperledger/besu/ethereum/trie/verkle/node/NullNode.java [hunk 1]
```diff
@@ -115,12 +126,44 @@ public String toDot(Boolean showRepeatingEdges) {
...
+        + getLocation().orElse(Bytes.EMPTY)
+        + "\"]\n";
+  }
+
+  @Override
+  public String getName() {
+    return isLeaf() ? "NullLeafNode" : "NullNode";
+  }
+
+  @SuppressWarnings("rawtypes")
```
Comment (matkt) [inline src/main/java/org/hyperledger/besu/ethereum/trie/verkle/node/NullNode.java:141]: for NullNode class maybe better to have the @SuppressWarnings directly for the class ? and not for each method

File: src/main/java/org/hyperledger/besu/ethereum/trie/verkle/node/NullNode.java [hunk 2]
```diff
@@ -115,12 +126,44 @@ public String toDot(Boolean showRepeatingEdges) {
...
+  @SuppressWarnings("unchecked")
+  public static <T> NullNode<T> nullLeafNode() {
+    if (nullLeafNode == null) {
+      nullLeafNode = new NullNode<>();
+      nullLeafNode.isLeaf = true;
+    }
+    return nullLeafNode;
+  }
+
+  public static <T> NullNode<T> nullLeafNode(Optional<T> previousValue) {
```
Comment (matkt) [inline src/main/java/org/hyperledger/besu/ethereum/trie/verkle/node/NullNode.java:164]: maybe better to rename the method to createNullLeafNode or newNullLeafNode

**Review by matkt — APPROVED** [review]
LGTM just added small comments

File: src/main/java/org/hyperledger/besu/ethereum/trie/verkle/node/NullNode.java (hunk 1 above)
Comment (lu-pinto) [reply src/main/java/org/hyperledger/besu/ethereum/trie/verkle/node/NullNode.java:141]: If we make it class wide then other `accept` methods might not report relevant warnings and they could be useful. I prefer to keep it locally to where we actually need to disable them

**Pull request merged**